clusterloader2/pkg/measurement/common/service_creation_latency.go

```diff
@@ -361,21 +493,33 @@ func (p *pingChecker) run() {
 			}
 			var ips []string
 			var port int32
-			switch p.svc.Spec.Type {
-			case corev1.ServiceTypeClusterIP:
-				ips = p.svc.Spec.ClusterIPs
-				port = p.svc.Spec.Ports[0].Port
-			case corev1.ServiceTypeNodePort:
-				ips = []string{pod.Status.HostIP}
-				port = p.svc.Spec.Ports[0].NodePort
-			case corev1.ServiceTypeLoadBalancer:
-				for _, ingress := range p.svc.Status.LoadBalancer.Ingress {
-					ips = append(ips, ingress.IP)
+			isIngress := p.ingress != nil
+			if isIngress {
```

Comment:
This is getting very nested and many ifs for ingress.
It would be more readable and easier to follow if they are decoupled.
Maybe one if in the beginning and then the rest can be generalized.

Reply:
Agree, improved readability there